app/scripts/modules/core/src/api/mock/mockHttpUtils.ts

```diff
@@ -0,0 +1,40 @@
+import { $httpBackend } from 'ngimport';
+
+export type Verb = 'GET' | 'POST' | 'PUT' | 'PATCH' | 'DELETE';
+export type UrlArg = string | RegExp | ((url: string) => boolean);
+
+export interface IDeferred {
+  promise: Promise<any>;
+  resolve: (result: any) => void;
+  reject: (error: any) => void;
+  settled: boolean;
+}
+
+export function deferred() {
+  const deferredObj = {} as IDeferred;
+  deferredObj.promise = new Promise((resolve, reject) => {
+    deferredObj.settled = false;
+
+    deferredObj.resolve = (result) => {
+      deferredObj.settled = true;
+      resolve(result);
+    };
+
+    deferredObj.reject = (error) => {
+      deferredObj.settled = true;
+      reject(error);
+    };
+  });
+
+  return deferredObj;
+}
+
+export function kickAngularJS() {
+  try {
+    $httpBackend.flush();
+    // eslint-disable-next-line no-empty
+  } catch (ignore) {}
+}
+
+export const tick = (ms?: number) => new Promise((resolve) => setTimeout(resolve, ms));
+export const isSuccessStatus = (status: number) => status >= 200 && status <= 299;
```

Comment:
What about 3xx? Do we generally expect the actual HTTP client implementations to automatically follow redirects? Presumably, we don't make use of 304 (Not Modified)?

Reply:
Great question, TBH I'm not sure about how all the codes are handled, but I believe that the most common 302 is handled natively by the browser.   We also don't do any caching for API calls -- deck never handles 304 responses directly, but the browser does natively for JS/etc assets.